src/current/v24.3/wal-failover.md

```diff
@@ -0,0 +1,495 @@
+
+No: If the side disk for WAL failover was **not** pre-configured before the cluster was started, changing the value of the [`storage.wal_failover.unhealthy_op_threshold`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-storage-wal-failover-unhealthy-op-threshold) cluster setting to something less than the [`storage.max_sync_duration`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-storage-max-sync-duration) cluster setting **will not have any effect and when the disk stalls, WAL failover will not trigger**. In other words, you cannot enable WAL failover at runtime for a single store cluster if you did not [pre-configure a side disk](#single-store-config).
+
+The recommended default value of the [`storage.wal_failover.unhealthy_op_threshold`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-storage-wal-failover-unhealthy-op-threshold) cluster setting is `100ms`.
+
+### 9. Can I change the log buffering on a node before enabling WAL failover?
+
+Yes.
+
+### 10. Does configuring WAL Failover on a running cluster require rolling restarts?
+
+Yes: For a single-store cluster, if you did not have a side disk pre-configured for WAL failover, you will need to do a rolling restart.
+
+Yes: For a multi-store cluster, if you are using file-based logging and did not have log buffering configured according to [3. Log Configuration for WAL Failover](#3-log-configuration-for-wal-failover), you will need to do a rolling restart.
+
+No: In all other cases, you can enable WAL failover without doing a rolling restart.
+
+### 11. After WAL failover is enabled on a cluster, is it safe for the customer to perform all normal operations?
+
+Yes.
+
+### 12. Can I change the value of the `storage.max_sync_duration` cluster setting at cluster runtime?
+
+Yes. To change [`storage.max_sync_duration`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-storage-max-sync-duration), issue a statement like the following:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET CLUSTER SETTING storage.max_sync_duration = '${desired_duration}';
+~~~
+
+### 13. If store A has a momentary stall, and it fails over to store B, and B's stall is longer than A's stall, will the WAL failback to A?
+
+Yes, the WAL will failback to store _A_ as soon as CockroachDB determines that _A_'s stall has abated (regardless of whether store _B_'s disk has stalled or not).
+
+### 14. If there are more than 2 stores, will the WAL failover cascade from store A to B to C?
+
+Yes, for example if store _A_'s disk stalls for `30s`, the WAL will failover to store B after `100ms`. While store _A_ is still stalled at the `20s` mark, if store _B_'s disk fails, store _B_ will failover to store _C_. When _B_ fails, only _B_'s WAL write will failover to _C_; if _A_ is still down, _A_'s write will not failover to _C_.
```

Comment:
This is confusing. I would not say this is a "yes" to the question. Store A will failover to store B, store B to store C, and store C to store A, but store A will never failover to store C.

Reply:
i replaced this sentence with what you wrote and the section now reads as follows, PTAL

> Store _A_ will failover to store _B_, store _B_ will failover to store _C_, and store _C_ will failover to store _A_, but store A will never failover to store C.

> However, the WAL failback operation will not cascade back until **all drives are available** - that is, if store _A_'s disk unstalls while store _B_ is still stalled, store _C_ will not failback to store _A_ until _B_ also becomes available again. In other words, _C_ must failback to _B_, which must then failback to _A_.